packages/cosmic-swingset/src/launch-chain.js

```diff
@@ -51,6 +54,22 @@ import { parseLocatedJson } from './helpers/json.js';
 const console = anylogger('launch-chain');
 const blockManagerConsole = anylogger('block-manager');
 
+/**
+ * @param {{ info: string } | null | undefined} upgradePlan
+ * @param {string} prefix
+ */
+const parseCoreProposalsFromUpgradePlan = (upgradePlan, prefix = '') => {
+  const { info: upgradeInfoJson = null } = upgradePlan || {};
+
+  const upgradePlanInfo =
+    upgradeInfoJson &&
+    parseLocatedJson(upgradeInfoJson, `${prefix}upgradePlan.info`);
+
+  // Handle the planned core proposals as just another action.
+  const { coreProposals } = upgradePlanInfo || {};
+  return harden({ coreProposals, info: upgradePlanInfo });
```

Comment:
Why extract `coreProposals` from the `info` here if we're gonna return `info` as well? Why not just return `info`? Or is it for the type?
```suggestion
  return harden(upgradePlanInfo || {});
```